contrib/mesos/pkg/executor/executor.go

```diff
@@ -388,53 +380,21 @@ func (k *Executor) LaunchTask(driver bindings.ExecutorDriver, taskInfo *mesos.Ta
 	// (a) we ignore future launches for it
 	// (b) we have a record of it so that we can kill it if needed
 	// (c) we're leaving podName == "" for now, indicates we don't need to delete containers
+	var launchTimer *time.Timer
+	if k.launchGracePeriod > 0 {
+		launchTimer = time.NewTimer(k.launchGracePeriod)
+	} else {
+		launchTimer = &time.Timer{}
+	}
 	k.tasks[taskId] = &kuberTask{
 		mesosTaskInfo: taskInfo,
+		launchTimer:   launchTimer,
```

Comment:
yes, that's a possibility. the earlier code refactoring didn't allow me to
do that. i may be able to now

On Wed, Dec 9, 2015 at 11:35 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/17917#discussion_r47114853>
> :
>
> >  	k.tasks[taskId] = &kuberTask{
> >  		mesosTaskInfo: taskInfo,
> > +		launchTimer:   launchTimer,
>
> Maybe easier to store the timer channel here which can also be nil and
> easily usable in a uniform way for nil and non-nil.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17917/files#r47114853>.
>